packages/react/src/user-preferences/userPreferences.ts

```diff
@@ -0,0 +1,51 @@
+export type UserPreferencesData = {
+  theme?: string;
+  lang?: string;
+};
+
+export type PreferencesUpdateCallback = (data?: UserPreferencesData) => void;
+
+export class UserPreferences {
```

Comment:
I need you to create a sort of sandbox system just like we did with Authentication packages:
- @microapp-ai/auth
- @microapp-ai/react

The goal is to enable developers to mock value on their local machines.
We created an Adapter strategy to achieve that. Take a look at the Authentication files.